test_server.py

```diff
@@ -43,36 +70,40 @@ def test_get_product(self):
 
     def test_query_product_list(self):
         self.setup_test_by_attribute('category', 'entertainment')
-        self.setup_test_by_attribute('discontinued', True)    
+        self.setup_test_by_attribute('discontinued', True)  
+        self.setup_test_by_attribute('price', '999')  
 
     def test_create_product(self):
         # save the current number of products for later comparison
         product_count = self.get_product_count()
         # add a new product
-        new_product = {'name': 'toaster', 'category': 'kitchen appliances'}
+        new_product = {'name': 'toaster', 'category': 'kitchen appliances', 'price': '99'}
         data = json.dumps(new_product)
         resp = self.app.post('/products', data=data, content_type='application/json')
         self.assertTrue( resp.status_code == HTTP_201_CREATED )
         new_json = json.loads(resp.data)
         self.assertTrue (new_json['name'] == 'toaster')
-        # Check default value of 'discontinued' is 'False'
+        # check default value of 'discontinued' is 'False'
         self.assertTrue (new_json['discontinued'] == False)
+        # check the price of the product
+        self.assertTrue (new_json['price'] == '99')
         # check that count has gone up and includes sammy
-        resp = self.app.get('/products')
+        resp = self.app.get('/products?limit=' + str(product_count + 1))
```

Comment:
Our default cutoff is '2'; should we choose to make is higher, than the 'limit' attribute is not needed.